common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -798,153 +794,539 @@ func (d *cassandraPersistence) CreateWorkflowExecution(
 		return nil, err
 	}
 
-	switch request.Mode {
-	case p.CreateWorkflowModeZombie:
-		// noop
-
-	default:
-		if err := createOrUpdateCurrentExecution(
-			batch,
-			request.Mode,
-			d.shardID,
-			domainID,
-			workflowID,
-			runID,
-			executionInfo.State,
-			executionInfo.CloseStatus,
-			executionInfo.CreateRequestID,
-			startVersion,
-			lastWriteVersion,
-			request.PreviousRunID,
-			request.PreviousLastWriteVersion,
-		); err != nil {
-			return nil, err
-		}
+	currentWorkflowWriteReq, err := d.prepareCurrentWorkflowForCreateWorkflowTxn(domainID, workflowID, runID, executionInfo, lastWriteVersion, request)
+	if err != nil {
+		return nil, err
 	}
 
-	if err := applyWorkflowSnapshotBatchAsNew(
-		batch,
-		d.shardID,
-		&newWorkflow,
-	); err != nil {
+	execution, err := d.prepareWorkflowExecutionForCreateWorkflowTxn(
+		executionInfo, versionHistories, checkSum,
+		nowTimestamp, lastWriteVersion,
+	)
+	if err != nil {
 		return nil, err
 	}
 
-	batch.Query(templateUpdateLeaseQuery,
-		request.RangeID,
-		d.shardID,
-		rowTypeShard,
-		rowTypeShardDomainID,
-		rowTypeShardWorkflowID,
-		rowTypeShardRunID,
-		defaultVisibilityTimestamp,
-		rowTypeShardTaskID,
-		request.RangeID,
-	)
+	activityInfos, err := d.prepareActivityInfosForWorkflowTxn(newWorkflow.ActivityInfos)
```

Comment:
Do we need this in create workflow operation?

Reply:
addressed offline. This is probably not being used but I have to keep the same behavior for refactoring.